test/collect_tests.jl

```diff
@@ -178,6 +178,17 @@
             obtainer = deepcopy,
         )
         @test model_data[1, :deep].data[1] < model_data[end, :deep].data[1]
+        agent_data, model_data = run!(
+            model,
+            agent_step!,
+            model_step!,
+            20;
+            agents_first=false,
+            when_model = [1, 20],
+            when = false,
+            mdata = [:flag, :year],
+            adata = [(:weight, mean)],
+        )
```

Comment:
These lines don't do any testing. The `@test` macro is the point at which a test is completed. A good test for this would be some model that counts agents with some flag = true in the model step, none of which are turned on until the first agent step. The test would be to run the model twice, one with agent, model stepping, then with model, agent stepping and compare the output.